test/e2e/utils.go

```diff
@@ -76,11 +79,45 @@ func prepare(t *testing.T) (*framework.TestCtx, string) {
 	namespace, err := ctx.GetNamespace()
 	Expect(err).ToNot(HaveOccurred())
 
-	err = e2eutil.WaitForOperatorDeployment(t, framework.Global.KubeClient, namespace, "nmstate-manager", 1, time.Second*5, time.Second*30)
+	err = WaitForOperatorDaemonSet(t, framework.Global.KubeClient, namespace, "nmstate-handler", time.Second*5, time.Second*90)
 	Expect(err).ToNot(HaveOccurred())
 	return ctx, namespace
 }
 
+// WaitForOperatorDeployment has the same functionality as WaitForDeployment but will no wait for the deployment if the
+// test was run with a locally run operator (--up-local flag)
+func WaitForOperatorDaemonSet(t *testing.T, kubeclient kubernetes.Interface, namespace, name string, retryInterval, timeout time.Duration) error {
+	return waitForDaemonSet(t, kubeclient, namespace, name, retryInterval, timeout, true)
+}
+
+func waitForDaemonSet(t *testing.T, kubeclient kubernetes.Interface, namespace, name string, retryInterval, timeout time.Duration, isOperator bool) error {
+	if isOperator && framework.Global.LocalOperator {
+		t.Log("Operator is running locally; skip waitForDeployment")
+		return nil
+	}
+	err := wait.PollImmediate(retryInterval, timeout, func() (done bool, err error) {
```

Comment:
What about using Gomega's Eventually? Then you just implement checkDaemonSet (to read it and check whether it's ready. The rest will be done using Eventually.

Reply:
This is copy paste from operators-sdk to wait for Deployments, my idea is to create a PR there in their framework with this, so I prefer to keep it like their code.